TLM/TLM/Manager/Impl/ExtCitizenInstanceManager.cs

```diff
@@ -1427,34 +1427,37 @@ bool citizenDebug
-        {
+                                          Vector3 startPos) {
 #if DEBUG
-            ref CitizenInstance citizenInstance = ref extInstance.instanceId.ToCitizenInstance();
-            
             bool citizenDebug =
-                (DebugSettings.CitizenId == 0 || DebugSettings.CitizenId == citizenInstance.m_citizen)
+                (DebugSettings.CitizenId == 0 || DebugSettings.CitizenId == instanceData.m_citizen)
                 && (DebugSettings.CitizenInstanceId == 0 || DebugSettings.CitizenInstanceId == instanceId)
-                && (DebugSettings.SourceBuildingId == 0 || DebugSettings.SourceBuildingId == citizenInstance.m_sourceBuilding)
-                && (DebugSettings.TargetBuildingId == 0 || DebugSettings.TargetBuildingId == citizenInstance.m_targetBuilding);
+                && (DebugSettings.SourceBuildingId == 0 || DebugSettings.SourceBuildingId == instanceData.m_sourceBuilding)
+                && (DebugSettings.TargetBuildingId == 0 || DebugSettings.TargetBuildingId == instanceData.m_targetBuilding);
 
             bool logParkingAi = DebugSwitch.BasicParkingAILog.Get() && citizenDebug;
-            // bool fineDebug = DebugSwitch.ExtendedParkingAILog.Get() && citizenDebug;
 
             if (logParkingAi) {
-                Log._Debug(
-                    $"ExtCitizenInstanceManager.IsAtOutsideConnection({extInstance.instanceId}): " +
-                    $"called. Path: {instanceData.m_path} sourceBuilding={instanceData.m_sourceBuilding}");
+                Log._Debug($"ExtCitizenInstanceManager.IsAtOutsideConnection({extInstance.instanceId}): " +
+                           $"called. Path: {instanceData.m_path} sourceBuilding={instanceData.m_sourceBuilding} targetBuilding={instanceData.m_targetBuilding}");
             }
 #else
             const bool logParkingAi = false;
 #endif
 
             ParkingAI parkingAiConf = GlobalConfig.Instance.ParkingAI;
+            float sqrMaxDistToPedestrianLane = parkingAiConf.MaxBuildingToPedestrianLaneDistance *
+                                               parkingAiConf.MaxBuildingToPedestrianLaneDistance;
+
             ref Building sourceBuilding = ref instanceData.m_sourceBuilding.ToBuilding();
+            bool ret = sourceBuilding.m_flags.IsFlagSet(Building.Flags.IncomingOutgoing) &&
+                       (startPos - sourceBuilding.m_position).sqrMagnitude <= sqrMaxDistToPedestrianLane;
 
-            bool ret = (sourceBuilding.m_flags & Building.Flags.IncomingOutgoing) != Building.Flags.None &&
-                (startPos - sourceBuilding.m_position).magnitude <= parkingAiConf.MaxBuildingToPedestrianLaneDistance;
+            if (!ret) {
+                ref Building targetBuilding = ref instanceData.m_targetBuilding.ToBuilding();
+                ret = targetBuilding.m_flags.IsFlagSet(Building.Flags.IncomingOutgoing) &&
+                      (startPos - targetBuilding.m_position).sqrMagnitude <= sqrMaxDistToPedestrianLane;
+            }
 
```

Comment:
[Note]
a bit of code duplication here. I wonder a local static method would be better:
```cs
static bool IsOutside(ref NetBuilding building, float sqrMaxDistToPedestrianLane) => building.m_flags.IsFlagSet(Building.Flags.IncomingOutgoing) && (startPos - building.m_position).sqrMagnitude <= sqrMaxDistToPedestrianLane;
return IsOutside(ref sourceBuilding, sqrMaxDistToPedestrianLane) || IsOutside(ref targetBuilding, sqrMaxDistToPedestrianLane);
```
Although I am not sure if old Mono would inline this for more performance.

Reply:
It won't inline anything. From multiple benchmarks I've created nothing gets inlined on Windows, on Linux/MacOS is more likely (you can notice that in form of crashes while patching things with Harmony).